### DISCUSSION TITLE
can't move /run/k3s/containerd to different dir (disk partition space issue)

### DISCUSSION THREAD

**Discussion by asloan7** [k3s-io/k3s#7611] [Q&A]
K3s Version:  1.25.9
OS: Linux (Centos, Rocky)

Our servers (not cloud) have partitions we can't change. We can add a big one (like /app or /usr/local/) to put all the working data, etc. in it. We need to be able to specify paths for all k3s things on startup so we don't fill up /var/lib/ and /run/. 
I use --data-dir /app/rancher/k3s
and possibly kubelet-arg: --data-dir=/usr/local/kubelet might work
but how can I move /run/k3s/ ?
It fills up to 429MB for a mostly empty deploy (after adding to rancher) and I'm sure would get large when 

Almost nothing else in /run is over 4kb, it probably isn't the appropriate place for gigs of containerd "rootfs"'s
(Path example: /run/k3s/containerd/io.containerd.runtime.v2.task/k8s.io/8705.../rootfs

**Expected behavior:**
Can specify where k3s/containerd lives

**Actual behavior:**
Can't specify where k3s/containerd lives
Workaround like setting up a symlink at /run/k3s to point to a dir on the big partition seems to get overwritten at k3s/k3s-agent service startup

PS.
k3s-killall.sh and k3s-killall-agent.sh will rm -rf /run/k3s so a symlink will also be lost if created there

**Comment by brandond** [comment]
`/run/k3s` should just contain ephemeral runtime data, which is why the killall script deletes it. On most distros there is tmpfs mounted at `/run`, so it shouldn't consume any actual storage off your root partition. Are you sure that it is consuming actual space on your nodes?

**Comment by asloan7** [comment]
Ok, not sure exactly how it works, but
/run/k3s "rootfs" mounts (potentially the big ones) map to "overlay" (which is on our big partition, due to --data-dir config?)
df -h example: 
overlay /run/k3s/containerd/io.containerd.runtime.v2.task/k8s.io/[id]/rootfs

kubelet/pods mounts are on a tmpfs (memory map???)
and there are /run/k3s "sandboxes" mapped to "shm" which don't take any space

I feel somewhat safe now :-)
